kani-compiler/src/kani_middle/reachability.rs

```diff
@@ -153,11 +153,7 @@ impl<'tcx> MonoItemsCollector<'tcx> {
     /// instruction looking for the items that should be included in the compilation.
     fn reachable_items(&mut self) {
         while let Some(to_visit) = self.queue.pop() {
-            // TODO: This should only check is_foreign_item() or even `has_body()`.
-            // We need https://github.com/rust-lang/rust/pull/118681 to land first.
-            if !self.collected.contains(&to_visit)
-                && !self.tcx.is_foreign_item(rustc_internal::internal(to_visit.clone()).def_id())
-            {
+            if !self.tcx.is_foreign_item(rustc_internal::internal(to_visit.clone()).def_id()) {
```

Comment:
Shoot, sorry I wasn't clear before and I totally missed this. This is what I meant:
```suggestion
            if !self.collected.contains(&to_visit) {
```